replication/event.go

```diff
@@ -543,6 +549,195 @@ func (e *GTIDEvent) OriginalCommitTime() time.Time {
+				},
+				{
+					Name:     "commit_group_ticket",
+					Optional: true,
+				},
+			},
+		},
+	}
+
+	err := serialization.Unmarshal(data, &e.msg)
+	if err != nil {
+		return err
+	}
+
+	f, err := e.msg.GetFieldByName("gtid_flags")
+	if err != nil {
+		return err
+	}
+	if v, ok := f.Type.(serialization.FieldIntFixed); ok {
+		e.CommitFlag = v.Value[0]
+	} else {
+		return errors.New("failed to get gtid_flags field")
+	}
+
+	f, err = e.msg.GetFieldByName("uuid")
+	if err != nil {
+		return err
+	}
+	if v, ok := f.Type.(serialization.FieldIntFixed); ok {
+		e.SID = v.Value
+	} else {
+		return errors.New("failed to get uuid field")
+	}
+
+	f, err = e.msg.GetFieldByName("gno")
+	if err != nil {
+		return err
+	}
+	if v, ok := f.Type.(serialization.FieldIntVar); ok {
+		e.GNO = v.Value
+	} else {
+		return errors.New("failed to get gno field")
+	}
+
+	f, err = e.msg.GetFieldByName("tag")
+	if err != nil {
+		return err
+	}
+	if v, ok := f.Type.(serialization.FieldString); ok {
+		e.Tag = v.Value
+	} else {
+		return errors.New("failed to get tag field")
+	}
+
+	f, err = e.msg.GetFieldByName("last_committed")
+	if err != nil {
+		return err
+	}
+	if v, ok := f.Type.(serialization.FieldIntVar); ok {
+		e.LastCommitted = v.Value
+	} else {
+		return errors.New("failed to get last_committed field")
+	}
+
+	f, err = e.msg.GetFieldByName("sequence_number")
+	if err != nil {
+		return err
+	}
+	if v, ok := f.Type.(serialization.FieldIntVar); ok {
+		e.SequenceNumber = v.Value
+	} else {
+		return errors.New("failed to get sequence_number field")
+	}
+
+	f, err = e.msg.GetFieldByName("immediate_commit_timestamp")
+	if err != nil {
+		return err
+	}
+	if v, ok := f.Type.(serialization.FieldUintVar); ok {
+		e.ImmediateCommitTimestamp = v.Value
+	} else {
+		return errors.New("failed to get immediate_commit_timestamp field")
+	}
+
+	f, err = e.msg.GetFieldByName("original_commit_timestamp")
+	if err != nil {
+		return err
+	}
+	if v, ok := f.Type.(serialization.FieldUintVar); ok {
+		e.OriginalCommitTimestamp = v.Value
+	} else {
+		return errors.New("failed to get original_commit_timestamp field")
+	}
+
+	f, err = e.msg.GetFieldByName("immediate_server_version")
+	if err != nil {
+		return err
+	}
+	if v, ok := f.Type.(serialization.FieldUintVar); ok {
+		e.ImmediateServerVersion = uint32(v.Value)
+	} else {
+		return errors.New("failed to get immediate_server_version field")
+	}
+
+	f, err = e.msg.GetFieldByName("original_server_version")
+	if err != nil {
+		return err
+	}
+	if v, ok := f.Type.(serialization.FieldUintVar); ok {
+		e.OriginalServerVersion = uint32(v.Value)
+	} else {
+		return errors.New("failed to get original_server_version field")
+	}
+
+	f, err = e.msg.GetFieldByName("transaction_length")
+	if err != nil {
+		return err
+	}
+	if v, ok := f.Type.(serialization.FieldUintVar); ok {
+		e.TransactionLength = v.Value
+	} else {
+		return errors.New("failed to get transaction_length field")
+	}
+
+	// TODO: add and test commit_group_ticket
```

Comment:
TODO

Reply:
I haven't yet been able to generate any events with an actual `commit_group_ticket`.

I first tried with `binlog_group_commit_sync_delay=...` but that didn't make any difference.

Then looking at the code I noticed that this has to do with Group Replication (part of InnoDB Cluster), So I [setup](https://dev.mysql.com/doc/mysql-shell/8.0/en/setting-up-innodb-cluster-and-mysql-router.html) a cluster with 3 sandbox instances... but that also didn't give me any events for testing.

Looks like this might also be known as [BGC](https://dev.mysql.com/doc/dev/mysql-server/latest/classbinlog_1_1BgcTicket.html#details) (Binlog Group Commit) tickets.